mplc/utils.py

```diff
@@ -126,3 +128,73 @@ def init_result_folder(yaml_filepath, cfg):
 
     logger.info("Result folder initiated")
     return cfg
+
+
+def init_gpu_config():
+    gpus = tf.config.experimental.list_physical_devices("GPU")
+    if gpus:
+        logger.info(f"Found GPU: {gpus[0].name}")
+        try:  # catch error when used on virtual devices
+            tf.config.experimental.set_memory_growth(gpus[0], True)
+        except ValueError as e:
+            logger.warning(str(e))
+        tf.config.experimental.set_virtual_device_configuration(
+            gpus[0],
+            [tf.config.experimental.VirtualDeviceConfiguration(memory_limit=constants.GPU_MEMORY_LIMIT_MB)]
+        )
+    else:
+        logger.info("No GPU found")
+
+
+def get_config_from_file(config_filepath):
+    config = load_cfg(config_filepath)
+    config = init_result_folder(config_filepath, config)
+
+    return config
+
+
+def parse_command_line_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-f", "--file", help="input config file")
+    parser.add_argument("-v", "--verbose", help="verbose output", action="store_true")
+    args = parser.parse_args()
+
+    return args
+
+
+def init_logger(debug=False):
+    logger.remove()
+    global log_filter
+    log_filter = MyFilter("INFO")
+    logger.opt(depth=1)
+    # Forward logging to standard output
+    logger.add(sys.stdout, enqueue=True, filter=log_filter, level=0)
+    if debug:
+        log_filter.set_to_debug_level()
+    else:
+        log_filter.set_to_info_level()
+
+
+class MyFilter:
+
+    def __init__(self, level):
+        self.level = level
+
+    def set_to_debug_level(self):
+        self.level = "DEBUG"
+
+    def set_to_info_level(self):
+        self.level = "INFO"
+
+    def __call__(self, record):
+        levelno = logger.level(self.level).no
+        return record["level"].no >= levelno
+
+
+def set_log_file(path):
+    logger.remove()
```

Comment:
Why do you need this line here ?
I will remove the handler added on line 171

Reply:
If you call this function twice, for instance when you change the folder for a new scenario, you will add the new log files. Not replace the old ones. 
So at the end, you will write on every file you set up. 
With this line, I remove every handler, (and I instantly put back the default handler) 
An other solution would have been saving the handler id, and removing it. But where could be stored the id, since we are on a library context ? 
So I choose to remove all handlers. 
At the ends, since a library is not really supposed to log anything, I make something the simplest as I could, which matches the main.py.